pkg/image/source_test.go

```diff
@@ -104,57 +104,67 @@ func TestDetectSource(t *testing.T) {
 		},
 		{
 			name:             "oci-tar-path",
+			fs:               getDummyTar(t, "a-potential/path", "oci-layout"),
 			input:            "a-potential/path",
 			source:           OciTarballSource,
 			expectedLocation: "a-potential/path",
-			tarPath:          "a-potential/path",
-			tarPaths:         []string{"oci-layout"},
 		},
 		{
 			name:             "unparsable-existing-path",
+			fs:               getDummyTar(t, "a-potential/path"),
 			input:            "a-potential/path",
 			source:           UnknownSource,
 			expectedLocation: "",
-			tarPath:          "a-potential/path",
-			tarPaths:         []string{},
 		},
 		// honor tilde expansion
 		{
 			name:             "oci-tar-path",
+			fs:               getDummyTar(t, "~/a-potential/path", "oci-layout"),
 			input:            "~/a-potential/path",
 			source:           OciTarballSource,
 			expectedLocation: "~/a-potential/path",
-			tarPath:          "~/a-potential/path",
-			tarPaths:         []string{"oci-layout"},
 		},
 		{
 			name:             "oci-tar-path-explicit",
+			fs:               getDummyTar(t, "~/a-potential/path", "oci-layout"),
 			input:            "oci-archive:~/a-potential/path",
 			source:           OciTarballSource,
 			expectedLocation: "~/a-potential/path",
-			tarPath:          "~/a-potential/path",
-			tarPaths:         []string{"oci-layout"},
 		},
 		{
 			name:             "oci-tar-path-with-scheme-separator",
+			fs:               getDummyTar(t, "a-potential/path:version", "oci-layout"),
 			input:            "a-potential/path:version",
 			source:           OciTarballSource,
 			expectedLocation: "a-potential/path:version",
-			tarPath:          "a-potential/path:version",
-			tarPaths:         []string{"oci-layout"},
 		},
 		{
 			name:             "singularity-path",
-			input:            "singularity:~/path/to/file.sif",
+			fs:               getDummySIF(t, "~/a-potential/path.sif"),
+			input:            "singularity:~/a-potential/path.sif",
 			source:           SingularitySource,
-			expectedLocation: "~/path/to/file.sif",
+			expectedLocation: "~/a-potential/path.sif",
+		},
+		{
+			name:             "singularity-path-tilde",
+			fs:               getDummySIF(t, "~/a-potential/path.sif"),
+			input:            "~/a-potential/path.sif",
+			source:           SingularitySource,
+			expectedLocation: "~/a-potential/path.sif",
+		},
+		{
+			name:             "singularity-path-explicit",
+			fs:               getDummySIF(t, "~/a-potential/path.sif"),
+			input:            "singularity:~/a-potential/path.sif",
+			source:           SingularitySource,
+			expectedLocation: "~/a-potential/path.sif",
 		},
```

Comment:
Is this the exact same test as `singularity-path` above?

Reply:
They are similar, but they test different functions (`detectSource` vs `detectSourceFromPath`.) `detectSource` calls `detectSourceFromPath` in some instances (when the source scheme is omitted), but not in others. For that reason, I think it's _probably_ best to test both explicitly, as is done with the other source types?